src/mlir/generator.cc

```diff
@@ -28,6 +28,29 @@ namespace
     region->push_back(new mlir::Block());
     return &region->back();
   }
+
+  /// Extracts a class type from join/meet
+  mlir::verona::ClassType extractClassType(mlir::Type type)
```

Comment:
I would sooner you return unknown if you encounter meet or join in the first cut.  You are trying to implement a form of type checking, and I think that should be done later, when we actually have information.

Reply:
This is not about type checking, but getting the class for the field access. If I leave it unknown, then I can't know if the type is of a class and which field it is.

I'm not sure how this would work, if at all:
```
class C { a }
class D { b }
foo(x: C | D) { return x.a }
```
but I'm expecting this is invalid syntax.

So, for now, I'm only supporting the cases where there's only one class.

I could add an error if there are more than one class type in the collection?

Reply:
That is not a syntax error.  It is type checking error.  You need to pass it through to type checking.  
```
class A 
{
  a
  b
}
class B 
{
  a
  c
}
foo(x: A|B) { return x.a }
```
Should type check. So unless you want to implement type checking either in this pass, or on the AST (which will also involve inference).  You are just going to have to let a later pass check the types.

Reply:
So, should this pass the initial lowering, too?
```
class C { a }
class D { b }
foo(x: C | D) { return x.a }
bar() { let d: D = new D; foo(d) }
```
but fail some later type check?

Then we'll have to change the validation on all dialect operations to not care at all about anything and let the type check pass to do all the right things and make sure none of the passes after the type check can ever add unknown types.

Reply:
Yes the type check for `foo` will fail.  But this could be pretty complex given how we want to infer generic parameters.

```
class C { a }
make[X](): X{ ... }
use_C(x: C) { ... }

bar()
{
  var x = make(); // Generic to be inferred based on use
  x.a = ...;
  use_C(x); // Constrains x to be C. Inference will back propagate this.
}
```
Unknown is all you can do unless you implement the inference algorithm.